packages/eslint-plugin/docs/rules/no-extraneous-return-types.mdx

```diff
@@ -0,0 +1,67 @@
+---
+description: 'Disallow extraneous union types in return type annotations.'
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+> 🛑 This file is source code, not the primary documentation location! 🛑
+>
+> See **https://typescript-eslint.io/rules/no-extraneous-return-types** for documentation.
+
+Explicit return type annotations are optional in TypeScript. However, explicit return types do make it visually more clear what type is returned by a function.
+
+A function can include return type annotations that are never used. This can happen when refactoring code or by mistake. Having unused return types adds unnecessary complexity because it needs to be handled by code that calls the function.
+
+This rule reports cases where a function has an unused return type annotation.
```

Comment:
[Docs] Well, they're _often_ optional, but sometimes required - e.g. for well-typed recursive functions. 

Also, this never actually mentions union types. Which makes it seem like the rule checks entire return type annotations for being necessary - not just individual unions. I think the explanation would be most helpful if it centered around that.

WDYT about something roughly like this?

```suggestion
Explicit function return type annotations can sometimes be helpful in TypeScript to describe a more precise return type than what the type checker would infer.
Functions that can return one of multiple types of values generally use union types to describe their return type.
This lint rule checks for "extraneous", or unnecessary, constituents of union type returns: those that don't describe a type the function can return.
```

...that's kind of clunky and I don't love it. Someone like @Josh-Cena who's good at describing things could probably do a better job. 😄

Reply:
I really like the way you've explained this 👍